### PR TITLE
Initial team prompt db schema

### PR DESCRIPTION
**Pull request by BartoszJarocki**
Implements #5916 

Scope

- [ ] Add new type teamPrompt to `type MeetingTypeEnum = 'poker' | 'retrospective' | 'action'`
- [ ] Add new table in PG TeamPromptResponse with fields id, createdAt, updatedAt, creatorId, meetingId, content, reactjis
- [ ] Change the Discussion table add new enum field teamPromptResponse to discussionTopicType enum

Important: `recurrence` is still being discussed and it's out of scope in this phase 

GQL types will be implemented in https://github.com/ParabolInc/parabol/issues/5917

### CODE REVIEW

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    CREATE TYPE "MeetingTypeEnum" AS ENUM (
+      'action',
+      'retrospective',
+      'poker'
+    );
+    ALTER TABLE "Team"
+      ALTER COLUMN "lastMeetingType" DROP DEFAULT,
+      ALTER COLUMN "lastMeetingType" TYPE "MeetingTypeEnum" USING "lastMeetingType"::text::"MeetingTypeEnum",
+      ALTER COLUMN "lastMeetingType" SET DEFAULT 'retrospective';
+    DROP TYPE "MeetingTypeEnum_delete";
```
Comment (BartoszJarocki) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:65]: reverting changes to enums is such a pain. I haven't found anything better but if there's a way, I'm happy to improve

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+      CREATE TABLE IF NOT EXISTS "TeamPromptResponse" (
+        "id" VARCHAR(50) PRIMARY KEY,
+        "createdAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        "updatedAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        "meetingId" VARCHAR(100) NOT NULL,
+        "userId" VARCHAR(100) NOT NULL,
+        "content" TEXT NOT NULL,
+        "plainTextContent" TEXT NOT NULL,
+        "reactjis" "Reactji"[] NOT NULL DEFAULT array[]::"Reactji"[],
```
Comment (BartoszJarocki) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:26]: I think this is enough and we don't need a separate table for reactjis. if there's a use case that'd benefit from that, I'm happy to add it

File: packages/server/database/types/Meeting.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import generateUID from '../../generateUID'
 import GenericMeetingPhase from './GenericMeetingPhase'
 
-export type MeetingTypeEnum = 'poker' | 'retrospective' | 'action'
+export type MeetingTypeEnum = 'poker' | 'retrospective' | 'action' | 'teamPrompt'
```
Comment (BartoszJarocki) [inline packages/server/database/types/Meeting.ts:4]: should we remove this type and use PG-generated enum to keep one source of truth?

**Comment by BartoszJarocki** [comment]
@tianrunhe would you mind taking a look? thank you!

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import getPgConfig from '../getPgConfig'
+
+export async function up() {
+  const client = new Client(getPgConfig())
+  await client.connect()
+  await client.query(`
+    DO $$
+    BEGIN
+      IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'Reactji') THEN
+        CREATE TYPE "Reactji" AS (id text, userId text);
```
Comment (tianrunhe) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:11]: -1 should we also put foreign key constraint on the `userId` here?

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 2 above)
Comment (tianrunhe) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:26]: +1 suggestion: add one more column `sortOrder` as v1

**Review by tianrunhe — CHANGES_REQUESTED** [review]
I like we move fast with pieces we can build today, while waiting for discussion on other parts! Just a few small suggestions.

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import {Client} from 'pg'
+import getPgConfig from '../getPgConfig'
+
+export async function up() {
+  const client = new Client(getPgConfig())
+  await client.connect()
+  await client.query(`
+    DO $$
+    BEGIN
+      IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'Reactji') THEN
```
Comment (mattkrick) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:10]: +2 nice use of composite types!

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      ALTER TYPE "MeetingTypeEnum" ADD VALUE 'teamPrompt';
+
+      CREATE TABLE IF NOT EXISTS "TeamPromptResponse" (
+        "id" VARCHAR(50) PRIMARY KEY,
+        "createdAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        "updatedAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        "meetingId" VARCHAR(100) NOT NULL,
+        "userId" VARCHAR(100) NOT NULL,
+        "content" TEXT NOT NULL,
+        "plainTextContent" TEXT NOT NULL,
```
Comment (mattkrick) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:25]: +1 everywhere else we call this `plaintextContent` (lowercase T). all depends if plaintext is a single word or 2 words 😅

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'Reactji') THEN
+        CREATE TYPE "Reactji" AS (id text, userId text);
+      END IF;
+
+      ALTER TYPE "DiscussionTopicTypeEnum" ADD VALUE 'teamPromptResponse';
+
+      ALTER TYPE "MeetingTypeEnum" ADD VALUE 'teamPrompt';
+
+      CREATE TABLE IF NOT EXISTS "TeamPromptResponse" (
+        "id" VARCHAR(50) PRIMARY KEY,
```
Comment (mattkrick) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:19]: +1 curious why no `"id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY`
Was it no good for the polls table?

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 1 above)
Comment (mattkrick) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:65]: for downmigrations i just wouldn't do it. up migrations are guaranteed to run eventually, so if an enum has an extra value for a couple minutes it won't hurt anything.

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    );
+    ALTER TABLE "Team"
+      ALTER COLUMN "lastMeetingType" DROP DEFAULT,
+      ALTER COLUMN "lastMeetingType" TYPE "MeetingTypeEnum" USING "lastMeetingType"::text::"MeetingTypeEnum",
+      ALTER COLUMN "lastMeetingType" SET DEFAULT 'retrospective';
+    DROP TYPE "MeetingTypeEnum_delete";
+
+    DROP TABLE IF EXISTS "TeamPromptResponse";
+
+    DROP TYPE "Reactji";
```
Comment (mattkrick) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:69]: +1 IF EXISTS. sometimes things get wonky, so i just add that check for everything

**Review by mattkrick — CHANGES_REQUESTED** [review]

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 4 above)
Comment (BartoszJarocki) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:11]: @mattkrick sure! 

@tianrunhe I think it isn't possible to add foreign key constraints in composite types, I googled the internet and couldn't find anything 🤔  

```
The syntax is comparable to CREATE TABLE, except that only field names and types can be specified; no constraints (such as NOT NULL) can presently be included. Note that the AS keyword is essential; without it, the system will think a different kind of CREATE TYPE command is meant, and you will get odd syntax errors.
```
https://www.postgresql.org/docs/13/rowtypes.html#ROWTYPES-DECLARING

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 6 above)
Comment (BartoszJarocki) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:25]: yeah, I was actually wondering what's the correct way 😅 will fix!

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 7 above)
Comment (BartoszJarocki) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:19]: I focused too much on enums and composite types and actually didn't think too much about this, very much my bad, it should be exactly like in polls!

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 8 above)
Comment (BartoszJarocki) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:69]: good catch!

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 2 above)
Comment (BartoszJarocki) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:26]: i'm happy to add it but what do you think is the use case for it?

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 1 above)
Comment (BartoszJarocki) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:65]: as it's working and I already spend time on this I'll leave it, next time I'll skip it 👍🏻

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 4 above)
Comment (tianrunhe) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts]: That's totally fine! I wasn't sure if there's a way to do it either 😅

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 2 above)
Comment (tianrunhe) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts]: Could be used in response grid sorting. See [here](https://github.com/ParabolInc/parabol/issues/5965#issuecomment-1042116030). Although I haven't thought about this carefully to see if `updatedAt` can be used to achieve sorting.

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+      ALTER TYPE "MeetingTypeEnum" ADD VALUE 'teamPrompt';
+
+      CREATE TABLE IF NOT EXISTS "TeamPromptResponse" (
+        "id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+        "createdAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        "updatedAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        "meetingId" VARCHAR(100) NOT NULL,
+        "userId" VARCHAR(100) NOT NULL,
+        "content" TEXT NOT NULL,
+        "plaintextContent" TEXT NOT NULL,
```
Comment (mattkrick) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:25]: +1 don't hesitate to tell me no, but if `content` is stored as JSON, then `plaintextContent` could be GENERATED and we won't need to do that at the application layer

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+      ALTER TYPE "MeetingTypeEnum" ADD VALUE 'teamPrompt';
+
+      CREATE TABLE IF NOT EXISTS "TeamPromptResponse" (
+        "id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+        "createdAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        "updatedAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+        "meetingId" VARCHAR(100) NOT NULL,
+        "userId" VARCHAR(100) NOT NULL,
+        "content" TEXT NOT NULL,
```
Comment (BartoszJarocki) [inline packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:24]: This should be JSON

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 2 above)
Comment (BartoszJarocki) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts]: @tianrunhe thanks! considering recent discussions around `sortOrder` I guess it should be an int. feel free to correct me, thought!

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 9 above)
Comment (BartoszJarocki) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:26]: I haven't found any easy way to do it so I'll create a task for it and let's continue with doing that conversion in the application layer for now.

**Comment by BartoszJarocki** [comment]
@mattkrick may I ask you to check it again?

File: packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts (hunk 9 above)
Comment (mattkrick) [reply packages/server/postgres/migrations/1644936872423_addTeamPromptResponseTable.ts:26]: > any chance you know a simple way of converting json to text in pg?

i think we talked about this in the 1:1. it'd have to be a function that traversed the json recursively picking out the text property on text nodes & aggregating them. definitely worth holding off for now

> I was thinking about it and I have one more question, is there any benefit of storing the same data in a different format in DB? what's the use case for that field, in general?

rich text is hard to query against, read for debugging, & no good for searching if we even wanted to implement something like elastic search

**Comment by mattkrick** [comment]
LGTM!

Review time: 2 hours

**Pull request merged**